tests/handlers/test_register.py

```diff
@@ -521,25 +538,20 @@ def test_auto_create_auto_join_where_no_consent(self):
         #    * The server is configured to auto-join to a room
         # (and autocreate if necessary)
 
-        event_creation_handler = self.hs.get_event_creation_handler()
-        # (Messing with the internals of event_creation_handler is fragile
-        # but can't see a better way to do this. One option could be to subclass
-        # the test with custom config.)
-        event_creation_handler._block_events_without_consent_error = "Error"
-        event_creation_handler._consent_uri_builder = Mock()
-        room_alias_str = "#room:test"
-        self.hs.config.auto_join_rooms = [room_alias_str]
-
         # When:-
-        #   * the user is registered and post consent actions are called
+        #   * the user is registered
         user_id = self.get_success(self.handler.register_user(localpart="jeff"))
-        self.get_success(self.handler.post_consent_actions(user_id))
```

Comment:
I *think* this change makes sense! If they have consented, then why would they not be joined to the room? It is a bit hard to tell from the logic what is supposed to happen in this case though.

Reply:
:man_shrugging:

Any idea how this was passing before?

Reply:
lovely.